include/deal.II/base/symmetric_tensor.h

```diff
@@ -3544,6 +3544,121 @@ DEAL_II_HOST constexpr inline DEAL_II_ALWAYS_INLINE
   return result;
 }
 
+/**
+ * An enumeration to describe the desired symmetry of the
+ * symmetric fourth order tensor
+ */
+enum class SymmetryType
+{
+  /**
+   * Only minor symmetry of the form:
+   * $A_{ijkl}=A_{jikl}=A_{ijlk}=A_{jilk}$.
+   */
+  minor,
+  /** 
+   * Both minor and major symmetry of the form:
+   * Minor - $A_{ijkl}=A_{jikl}=A_{ijlk}=A_{jilk}$
+   * Major - $A_{ijkl}=A_{klij}$.
+   */
+  minor_major,
+};
+
+/**
+ * Return the symmetrized version of a full rank-4 tensor, i.e.
+ * as a symmetric rank-4 tensor.
+ * @param symmetry_type decides the type of symmetry.
+ * This is the version for general dimensions.
+ *
+ * @relatesalso SymmetricTensor
+ */
+template <int dim, typename Number>
+DEAL_II_HOST constexpr inline DEAL_II_ALWAYS_INLINE
+  SymmetricTensor<4, dim, Number>
+  symmetrize(const Tensor<4, dim, Number> &t,
+             const SymmetryType &symmetry_type = SymmetryType::minor)
```

Comment:
Have you considered not giving this argument a default value? It's not clear to me that one or the other choice for this argument is "more natural" and so should be the default. It might be better to just *always* require the caller to say what it is that they want, if only to force them to think about what that is (and thereby to also ensure that their choice is documented).

Reply:
Thanks for your comment. Now that I think about it, the current  design of SymmetricTensor class _requires_ minor symmetry by default. So the only additional symmetry we can have is the major symmetry. This means that we don't even need the proposed enum class TensorSymmetryType. We could simply have a bool argument for additional major symmetry instead of the TensorSymmetryType argument.

Reply:
Yes, I missed that too. This sounds like the right approach then!